processor-api/src/main/java/com/synopsys/integration/alert/processor/api/filter/extractor/ProjectNotificationExtractor.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * processor-api
+ *
+ * Copyright (c) 2021 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.processor.api.filter.extractor;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.google.gson.Gson;
+import com.synopsys.integration.alert.common.rest.model.AlertNotificationModel;
+import com.synopsys.integration.alert.processor.api.filter.model.FilterableNotificationWrapper;
+import com.synopsys.integration.blackduck.api.manual.component.ProjectNotificationContent;
+import com.synopsys.integration.blackduck.api.manual.enumeration.NotificationType;
+
+@Component
+public class ProjectNotificationExtractor extends NotificationExtractor<ProjectNotificationContent> {
+    @Autowired
+    public ProjectNotificationExtractor(Gson gson) {
+        super(NotificationType.PROJECT, ProjectNotificationContent.class, gson);
+    }
+
+    @Override
+    protected FilterableNotificationWrapper convertToFilterableNotificationWrapper(AlertNotificationModel alertNotificationModel, ProjectNotificationContent projectNotificationContent) {
+        return FilterableNotificationWrapper.project(alertNotificationModel, projectNotificationContent, projectNotificationContent.getProjectName());
```

Comment:
Can we modify FilterableNotificationWrapper further so that we don't need to pass AlertNotificationmodel and its subset of ProjectNotificationContent? I don't think these extractors will ever need AlertNotificationmodel

Reply:
I think we will need it for instantiating the `StatefulProvider` (or new equivalent) eventually.